File: app/src/main/java/org/carstenf/wordfinder/WordFinderPreferences.java
/*
 * Copyright Carsten Friedrich (Carsten.Friedrich@gmail.com)
 *
 * License: GNU GENERAL PUBLIC LICENSE 3.0 (https://www.gnu.org/copyleft/gpl.html)
 *
 */
package org.carstenf.wordfinder;

import android.os.Bundle;
import android.widget.FrameLayout;

import androidx.appcompat.app.AppCompatActivity;

public class WordFinderPreferences extends AppCompatActivity {

	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		// Create a FrameLayout to host the fragment
		FrameLayout frame = new FrameLayout(this);
		frame.setId(R.id.content);
		setContentView(frame);

		// Load the PreferenceFragment
		getSupportFragmentManager()
				.beginTransaction()
				.replace(R.id.content, new WordFinderSettingsFragment())
				.commit();
	}
}

File: app/src/main/kotlin/org/carstenf/wordfinder/WordFinderPreferences.kt
/*
 * Copyright Carsten Friedrich (Carsten.Friedrich@gmail.com)
 *
 * License: GNU GENERAL PUBLIC LICENSE 3.0 (https://www.gnu.org/copyleft/gpl.html)
 *
 */
package org.carstenf.wordfinder

import android.os.Bundle
import android.widget.FrameLayout
import androidx.appcompat.app.AppCompatActivity

class WordFinderPreferences : AppCompatActivity() {
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Enable the up button in the action bar
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setDisplayShowHomeEnabled(true)

        // Create a FrameLayout to host the fragment
        val frame = FrameLayout(this)
        frame.id = R.id.content
        setContentView(frame)

        // Load the PreferenceFragment
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.content, WordFinderSettingsFragment())
            .commit()

        frame.fitsSystemWindows = true
    }
    // Handle the up button press
    override fun onSupportNavigateUp(): Boolean {
        finish()
        return true
    }

}